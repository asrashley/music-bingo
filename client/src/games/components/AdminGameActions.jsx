import React from 'react';
import PropTypes from 'prop-types';
import { saveAs } from 'file-saver';

import {
  BusyDialog, ConfirmDialog, FileDialog, ProgressDialog
} from '../../components';

import { addMessage } from '../../messages/messagesSlice';
import { importDatabase } from '../../admin/adminSlice';
import {
  deleteGame, fetchGamesIfNeeded, importGame, invalidateGames
} from '../gamesSlice';

import { api } from '../../endpoints';

import { GamePropType } from '../types/Game';

export function AdminActionPanel({ deleteGame, exportGame, importGame, game }) {
  return (
    <div className="action-panel">
      <button className="btn btn-primary ml-2" onClick={importGame}>
        Import a game
      </button>
      {(game && exportGame) && <button className="btn btn-primary ml-2"
        onClick={exportGame}>Export game</button>}
      {(game && deleteGame) && <button className="btn btn-danger ml-2"
        onClick={deleteGame}>Delete game</button>}
    </div>
  );
}

AdminActionPanel.propTypes = {
  game: GamePropType,
  deleteGame: PropTypes.func,
  exportGame: PropTypes.func,
  importGame: PropTypes.func.isRequired,
};

function ErrorMessage({ error }) {
  if (!error) {
    return null;
  }
  return (
    <div className="alert alert-warning" role="alert">
      <span className="error-message">{error}</span>
    </div>
    );
}
ErrorMessage.propTypes = {
  error: PropTypes.string
};

export class AdminGameActions extends React.Component {
  static DATABASE_IMPORT = 1;
  static GAME_IMPORT = 2;

  static propTypes = {
    databaseImporting: PropTypes.object,
    dispatch: PropTypes.func.isRequired,
    onDelete: PropTypes.func.isRequired,
    game: GamePropType,
    gameImporting: PropTypes.object,
    importing: PropTypes.object,
  };

  state = {
    ActiveDialog: null,
    dialogData: null,
    importType: 0,
    error: null
  };

  render() {
    const { ActiveDialog, dialogData, error } = this.state;
    const { children, game, importing } = this.props;

    return (
      <React.Fragment>
        <ErrorMessage error={error} />
        {children}
        <AdminActionPanel
          game={game}
          deleteGame={this.confirmDelete}
          exportGame={this.exportGame}
          importGame={this.onClickImportGame}
        />;
        {ActiveDialog && <ActiveDialog backdrop {...dialogData} {...importing} /> }
      </React.Fragment>);
  }

  confirmDelete = (ev) => {
    ev.preventDefault();
    const { game } = this.props;
    this.setState({
      ActiveDialog: ConfirmDialog,
      dialogData: {
        changes: [
          `Delete group ${game.id}`
        ],
        title: "Confirm delete game",
        onCancel: this.cancelDialog,
        onConfirm: this.deleteGame,
      }
    });
    return false;
  };

  cancelDialog = () => {
    this.setState({
      ActiveDialog: null,
      dialogData: null,
      importType: 0,
    });
  };

  deleteGame = () => {
    const { game, dispatch, onDelete } = this.props;
    this.setState({ ActiveDialog: null, dialogData: null });
    dispatch(deleteGame(game)).then(result => {
      const { payload } = result;
      if (payload?.error || result.error) {
        this.setState({ error: (payload.error ? payload.error : result.error) });
      } else {
        onDelete(game);
      }
    });
  };

  onClickImportDatabase = () => {
    this.setState({
      ActiveDialog: FileDialog,
      dialogData: {
        title: 'Select a json database file to import',
        accept: '.json,application/json',
        submit: 'Import database',
        onCancel: this.cancelDialog,
        onFileUpload: this.onFileSelected
      },
      importType: AdminGameActions.DATABASE_IMPORT
    });
  };

  onChangeReplaceDate = (e) => {
    const { dialogData } = this.state;
    this.setState({
      dialogData: {
        ...dialogData,
        replaceDate: e.target.checked
      }
    });
  }

  onClickImportGame = () => {
    const extraFields = <div className="form-control">
      <label htmlFor="replaceDate" className="replace-date-label">Replace game start date during import?</label>
      <input type="checkbox" name="replaceDate" defaultChecked onChange={this.onChangeReplaceDate}></input>
    </div>;

    this.setState({
      ActiveDialog: FileDialog,
      dialogData: {
        title: 'Select a gameTracks.json file to import',
        accept: '.json,application/json',
        submit: 'Import game',
        onCancel: this.cancelDialog,
        onFileUpload: this.onFileSelected,
        replaceDate: true,
        extraFields
      },
      importType: AdminGameActions.GAME_IMPORT
    });
  };

  importComplete = () => {
    const { databaseImporting, gameImporting, dispatch } = this.props;
    const { importType } = this.state;
    if (importType === AdminGameActions.DATABASE_IMPORT && databaseImporting?.done === true) {
        document.location.reload();
        return;
    }
    if (importType === AdminGameActions.GAME_IMPORT && gameImporting?.done === true) {
      dispatch(invalidateGames());
      dispatch(fetchGamesIfNeeded());
    }
    this.setState({
      ActiveDialog: null,
      dialogData: null,
      importType: 0,
    });
  };

  exportDatabase = () => {
    const { dispatch } = this.props;
    this.setState(() => ({
      ActiveDialog: BusyDialog,
      dialogData: {
        onClose: this.cancelDialog,
        title: "Exporting database",
        text: "Please wait, exporting database...",
      }
    }), () => {
      dispatch(api.exportDatabase()).then(response => {
        const filename = `database-${Date.now()}.json`;
        return response.payload.blob().then(blob => {
          this.setState({ ActiveDialog: null, dialogData: null });
          saveAs(blob, filename);
          return filename;
        });
      });
    });
  };

  exportGame = () => {
    const { dispatch, game } = this.props;
    dispatch(api.exportGame({
      gamePk: game.pk
    })).then(response => {
      const filename = `game-${game.id}.json`;
      return response.payload.blob().then(blob => {
        saveAs(blob, filename);
        return filename;
      });
    });
  };

  onFileSelected = (file) => {
    var reader = new FileReader();
    reader.onload = (ev) => this.onFileLoaded(file.name, ev);
    reader.onerror = (err) => {
      this.setState({ activeDialog: null, activeDialogData: null });
      addMessage({ type: "error", text: err });
    };
    reader.readAsText(file);
  };

  onFileLoaded(filename, event) {
    const { dispatch } = this.props;
    const { importType } = this.state;
    const { replaceDate } = this.state.dialogData;
    const title = importType === AdminGameActions.GAME_IMPORT ?
      `Importing game from "${filename}"` :
      `Importing database from "${filename}"`;

    this.setState({
      ActiveDialog: ProgressDialog,
      dialogData: {
        title,
        onCancel: this.cancelDialog,
        onClose: this.importComplete
      }
    });
    const data = JSON.parse(event.target.result);
    if (importType === AdminGameActions.GAME_IMPORT) {
      if (replaceDate === true && 'Games' in data) {
        const start = new Date().toISOString();
        const end = new Date(Date.now() + 12*3600000).toISOString();
        data.Games = data.Games.map(game => ({
          ...game,
          start,
          end
        }));
      }
      dispatch(importGame(filename, data));
    } else {
      dispatch(importDatabase(filename, data));
    }
  }
}


