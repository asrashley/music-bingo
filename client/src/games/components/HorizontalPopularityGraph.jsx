import React from 'react';
import PropTypes from 'prop-types';

import { ThemeRow } from './ThemeRow';
import { ThemePropType } from '../types/Theme';

export function HorizontalPopularityGraph({ popularity, onRotate }) {
  return (
    <table className="horiz-popularity-graph">
      <thead>
        <tr>
          <th className="title">Theme</th>
          <th className="count">Popularity
            <button className="btn btn-light rotate-icon" onClick={onRotate}>&nbsp;</button>
          </th>
        </tr>
      </thead>
      <tbody>
        {popularity.map((theme, idx) => <ThemeRow key={idx} index={idx} theme={theme} />)}
      </tbody>
    </table>
  );
}

HorizontalPopularityGraph.propTypes = {
  popularity: PropTypes.arrayOf(ThemePropType).isRequired,
  onRotate: PropTypes.func.isRequired
};
