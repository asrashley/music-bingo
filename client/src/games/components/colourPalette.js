
export const colours = [
  '#FF0000',
  '#008000',
  '#00EFEF',
  '#808000',
  '#BFBF00',
  '#808080',
  '#008080',
  '#00EF00',
  '#C0C0C0',
  '#800000',
  '#0000FF',
  '#000080',
  '#FF00FF',
  '#800080',
];
