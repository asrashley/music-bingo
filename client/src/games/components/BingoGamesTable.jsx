import React from 'react';
import PropTypes from 'prop-types';
import { reverse } from 'named-urls';
import routes from '../../routes';
import { Link } from 'react-router-dom';

import { DateTime } from '../../components/DateTime';

import { GamePropType } from '../types/Game';
import { UserPropType } from '../../user/types/User';

const TableRow = ({ game, past, user }) => {
  let ticketColumn, themeColumn;
  let rowClass = `round-${game.round}`;

  if (past) {
    if (user.groups?.guests === true) {
      themeColumn = (<span>{game.title}</span>);
      ticketColumn = (<span>Log in to view track listing</span>);
    } else {
      const url = reverse(`${routes.trackListing}`, { gameId: game.id });
      themeColumn = (<Link to={url}>{game.title}</Link>);
      ticketColumn = (<Link to={url}>View track listing</Link>);
    }
  } else if (game.userCount === 0) {
    const url = reverse(`${routes.chooseTickets}`, { gameId: game.id });
    themeColumn = (<Link to={url}>{game.title}</Link>);
    ticketColumn = (<Link to={url}>Choose tickets</Link>);
  } else {
    const ticketUrl = reverse(`${routes.play}`, { gameId: game.id });
    const themeUrl = reverse(`${routes.chooseTickets}`, { gameId: game.id });
    themeColumn = (<Link to={ themeUrl } > { game.title }</Link>);
    ticketColumn = (<Link to={ticketUrl}>You have chosen {game.userCount} ticket{(game.userCount > 1) ? 's' : ''}</Link>);
  }
  if (game.options && game.options.colour_scheme) {
    rowClass += ` ${game.options.colour_scheme}-theme`;
  }
  const start = new Date(game.start);
  return (
    <React.Fragment>
      {game.round === 1 && (<tr>
        <th colSpan="4">{start.toDateString()}</th>
      </tr>)}
      <tr className={rowClass} data-testid={ `${past ? "past" : ""}game[${game.pk}]`}>
        <td className="round-column">{game.round}</td>
        <td className="date-column"><DateTime date={start} ampm={true} /></td>
        <td className="theme-column">{themeColumn}</td>
        <td className="ticket-column">{ticketColumn}</td>
      </tr>
    </React.Fragment>
  );
};
TableRow.propTypes = {
  game: GamePropType.isRequired,
  past: PropTypes.bool,
  user: UserPropType.isRequired
};

export const BingoGamesTable = ({ title, games, past, onReload, footer, user }) => {
  return (
    <table className="table table-bordered game-list">
      <thead>
        <tr>
          <th colSpan="4" className="available-games">{title}
            <button className="btn btn-light refresh-icon btn-sm" aria-label="Reload" onClick={onReload}>&#x21bb;</button>
          </th>
        </tr>
        <tr>
          <th className="round-column">Round</th>
          <th className="date-column">Date and Time</th>
          <th className="theme-column">Theme</th>
          {past && <th className="ticket-column">Track listing</th>}
          {!past && <th className="ticket-column">Your Tickets</th>}
        </tr>
      </thead>
      <tbody>
        {games.map((game, idx) => (<TableRow game={game} key={idx} past={past} user={user} />))}
      </tbody>
      {footer && <tfoot>{footer}</tfoot>}
    </table>
  );
};

BingoGamesTable.propTypes = {
  title: PropTypes.string.isRequired,
  games: PropTypes.arrayOf(GamePropType).isRequired,
  past: PropTypes.bool,
  onReload: PropTypes.func.isRequired,
  footer: PropTypes.node,
  user: UserPropType.isRequired
};
