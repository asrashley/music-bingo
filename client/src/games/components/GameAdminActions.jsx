import React from 'react';
import { useDispatch } from 'react-redux';
import { AdminActions } from '../../admin/components/AdminActions';
import { fetchGamesIfNeeded, invalidateGames } from '../gamesSlice';

export function GameAdminActions() {
    const dispatch = useDispatch();
    const onReload = () => {
        dispatch(invalidateGames());
        dispatch(fetchGamesIfNeeded());
    };

    return (
        <AdminActions alwaysShowChildren={true}>
            <button
                className="btn btn-primary"
                onClick={onReload}>
                Reload
            </button>
        </AdminActions>
    );
}
