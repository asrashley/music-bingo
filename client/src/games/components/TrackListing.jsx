import React from 'react';

import { formatDuration } from '../../components/DateTime';
import { GamePropType } from '../types/Game';
import { TrackPropType } from '../types/Track';

const TableRow = ({ game, track }) => {
  let rowClass = "track";
  if (game.options && game.options.colour_scheme) {
    rowClass += ` ${game.options.colour_scheme}-theme`;
  }

  return (
    <tr className={rowClass} data-testid={ `track[${track.pk}]`}>
      <td className="number">{track.number + 1}</td>
      <td className="start-time">{formatDuration(track.start_time)}</td>
      <td className="title">{track.title}</td>
      <td className="artist">{track.artist}</td>
      <td className="album">{track.album}</td>
      <td className="duration">{formatDuration(track.duration)}</td>
    </tr>

  );
};

TableRow.propTypes = {
  game: GamePropType,
  track: TrackPropType
};

export function TrackListing({ game }) {
  return (
    <div>
      <table className="table table-bordered track-listing">
        <thead>
          <tr>
            <th colSpan="6" className="heading">Track listing for Game {game.id}: "{game.title}"</th>
          </tr>
          <tr>
            <th className="number">#</th>
            <th className="start-time">Start</th>
            <th className="title">Title</th>
            <th className="artist">Artist</th>
            <th className="album">Album</th>
            <th className="duration">Duration</th>
          </tr>
        </thead>
        <tbody>
          {game.tracks.map((track, idx) => (<TableRow track={track} game={game} key={idx} />))}
        </tbody>
      </table>
    </div>
  );
}

TrackListing.propTypes = {
  game: GamePropType,
};
