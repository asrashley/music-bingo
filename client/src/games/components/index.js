import { ListGamesPage } from './ListGamesPage';
import { TrackListing } from './TrackListing';
import { TrackListingPage } from './TrackListingPage';
import { ModifyGame } from './ModifyGame';
import { PastGamesPage } from './PastGamesPage';

export {
  ListGamesPage,
  ModifyGame,
  PastGamesPage,
  TrackListing,
  TrackListingPage,
};
