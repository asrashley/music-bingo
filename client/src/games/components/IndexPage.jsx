import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { reverse } from 'named-urls';

import { initialState } from '../../app/initialState';
import { fetchUserIfNeeded, userIsLoggedIn } from '../../user/userSlice';
import { fetchGamesIfNeeded } from '../gamesSlice';
import routes from '../../routes';
import { LoginDialog } from '../../user/components/LoginDialog';

import '../styles/games.scss';

const utcFields = ['getUTCHours', 'getUTCMinutes',
  'getUTCSeconds', 'getUTCDate',
  'getUTCMonth', 'getUTCFullYear'];

const localFields = utcFields.map(f => f.replace('UTC', ''));


const DateTime = ({ date, useUTC=false, withTimezone=false, withDate=true, withTime=true, ...params }) => {
  if (typeof (date) === "string") {
    date = new Date(date);
  };
  let fields = (useUTC === false) ? localFields : utcFields;
  fields = fields.map((f) => {
    const val = date[f]();
    if (/Year/.test(f)) {
      return `${val}`;
    }
    if (/Month/.test(f)) {
      return `0${val + 1}`.slice(-2);
    }
    return `0${val}`.slice(-2);
  });
  let tz = '';
  if (withTimezone === true) {
    if (useUTC === false) {
      const offset = date.getTimezoneOffset();
      if (offset === 0) {
        tz = ' GMT';
      } else {
        tz = ' BST';
      }
    } else {
      tz = ' UTC';
    }
  }
  const time = withTime ? `${fields[0]}:${fields[1]}:${fields[2]}` : '';
  const dstr = withDate ? `${fields[3]}/${fields[4]}/${fields[5]}` : '';
  if (!withTime) {
    tz = '';
  }
  return `${time} ${dstr} ${tz}`;
};

const TableRow = ({ game }) => {
  let url, linkText;

  if (game.userCount === 0) {
    linkText = "Choose tickets";
    url = reverse(`${routes.game}`, { gamePk: game.pk });
  } else {
    url = reverse(`${routes.play}`, { gamePk: game.pk });
    linkText = `You have chosen ${game.userCount} ticket${(game.userCount > 1) ? 's' : ''}`;
  }
  return (
    <tr>
      <td className="round-column">{game.gameRound}</td>
      <td className="date-column">
        <DateTime date={game.start} withTime={false} />
      </td>
      <td className="theme-column">
        <Link to={reverse(`${routes.game}`, { gamePk: game.pk })}>{game.title}</Link>
      </td>
      <td className="ticket-column">
        <Link to={url}>{linkText}</Link>
      </td>
    </tr>
  );
};

class IndexPage extends React.Component {
  static propTypes = {
    dispatch: PropTypes.func.isRequired,
    loggedIn: PropTypes.bool.isRequired,
    user: PropTypes.object,
    games: PropTypes.array,
  };

  componentDidMount() {
    const { dispatch } = this.props;
    dispatch(fetchUserIfNeeded());
    dispatch(fetchGamesIfNeeded());
  }

  componentWillReceiveProps(nextProps) {
    const { user, dispatch } = nextProps;
    if (user.pk !== this.props.user.pk) {
      dispatch(fetchGamesIfNeeded());
    }
  }

  render() {
    const { games, user, loggedIn } = this.props;
    return (
      <div id="games-page">
        {user && <h2 className="greeting">Hello {user.username}</h2>}
        <table className="table table-striped table-bordered game-list">
          <thead>
            <tr>
              <th colSpan="4" className="available-games">Available Bingo games</th>
            </tr>
            <tr>
              <th className="round-column">Round</th>
              <th className="date-column">Date</th>
              <th className="theme-column">Theme</th>
              <th className="ticket-column">Your Tickets</th>
            </tr>
          </thead>
          <tbody>
            {games.map((game, idx) => (<TableRow game={game} key={idx} />))}
          </tbody>
        </table>
        {!loggedIn && <LoginDialog dispatch={this.props.dispatch} onSuccess={() => null} />}
      </div>
    );
  }
}

//IndexPage.whyDidYouRender = true;

const mapStateToProps = (state, ownProps) => {
  state = state || initialState;
  const { user } = state;
  const { games, order } = state.games;
  const { location } = ownProps;
  const gameList = [];
  order.forEach(pk => gameList.push(games[pk]));
  return {
    loggedIn: userIsLoggedIn(state),
    user,
    location,
    games: gameList,
  };
};

IndexPage = connect(mapStateToProps)(IndexPage);

export {
  IndexPage
};
