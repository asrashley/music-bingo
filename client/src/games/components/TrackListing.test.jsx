import React from 'react';
import { getByText } from '@testing-library/react';

import { renderWithProviders } from '../../testHelpers';
import { formatDuration } from '../../components/DateTime';
import { TrackListing } from './TrackListing';

describe('TrackListing component', () => {
  it('to render a game with 3 tracks', () => {
    const songs = [{
      "pk": 359,
      "filename": "01-48- You Spin Me Round (Like a Record) [Original 7 Mix].mp3",
      "title": "You Spin Me Round (Like a Record)",
      "duration": 30016,
      "channels": 2,
      "sample_rate": 44100,
      "sample_width": 16,
      "bitrate": 256,
      "uuid": "urn:uuid:0c76319e-65f1-5b4a-878c-bf06fe7be0ff",
      "directory": 78,
      "album": "100 Hits 80s Essentials",
      "artist": "Dead Or Alive"
    }, {
      "pk": 627,
      "filename": "15 Mad World (From 'Donnie Darko').mp3",
      "title": "Mad World (From \"Donnie Darko\")",
      "duration": 30016,
      "channels": 2,
      "sample_rate": 44100,
      "sample_width": 16,
      "bitrate": 256,
      "uuid": "urn:uuid:c7e63924-f46b-5557-a4d2-92ce83b5d21d",
      "directory": 86,
      "album": "Now That's What I Call Movies",
      "artist": "Tears For Fears"
    }, {
      "pk": 618,
      "filename": "12 Why (From 'Soup for One').mp3",
      "title": "Why (From \"Soup for One\")",
      "duration": 30016,
      "channels": 2,
      "sample_rate": 44100,
      "sample_width": 16,
      "bitrate": 257,
      "uuid": "urn:uuid:9b4557e6-ca6f-51f6-875c-15e2dc57a576",
      "directory": 86,
      "album": "Now That's What I Call Movies",
      "artist": "Carly Simon"
    }];
    const tracks = [{
      ...songs[0],
      "game": 1,
      "number": 1,
      "pk": 2,
      "start_time": 40072
    }, {
      ...songs[1],
      "game": 1,
      "number": 2,
      "pk": 3,
      "start_time": 71096
    }, {
      ...songs[2],
      "game": 1,
      "number": 3,
      "pk": 4,
      "start_time": 102120
      }];
    const game = {
      "pk": 2,
      "id": "18-04-25-2",
      "title": "Pot Luck",
      "start": "2018-04-05T20:43:00Z",
      "end": "2018-04-06T19:43:40Z",
      "options": {
        "colour_scheme": "green",
        "number_of_cards": 24,
        "include_artist": true,
        "page_size": "A4",
        "columns": 5,
        "rows": 3,
        "checkbox": false,
        "cards_per_page": 4,
        "backgrounds": ["#d9ffd9", "#f0fff0", "#d9ffd9", "#f0fff0", "#d9ffd9", "#f0fff0", "#d9ffd9", "#f0fff0", "#d9ffd9", "#f0fff0", "#d9ffd9", "#f0fff0", "#d9ffd9", "#f0fff0", "#d9ffd9"]
      },
      "userCount": 0,
      tracks,
      ticketOrder: [2, 3, 4],
      isFetchingDetail: false,
      invalidDetail: false,
      lastUpdated: 1675455186677,
      isModifying: false,
    };
    const result = renderWithProviders(<TrackListing game={game} />);
    result.getByText(game.title, { exact: false });
    tracks.forEach(track => {
      const tid = `track[${track.pk}]`;
      const dur = formatDuration(track.duration);
      expect('dur').not.toMatch(/N?aN?/);
      const row = result.getByTestId(tid);
      expect(row).toBeVisible()
      expect(row).toHaveClass(`${game.options.colour_scheme}-theme`);
      getByText(row, track.title);
      getByText(row, track.artist);
      getByText(row, track.album);
      getByText(row, dur);
    });
    expect(result.asFragment()).toMatchSnapshot();
  });
});