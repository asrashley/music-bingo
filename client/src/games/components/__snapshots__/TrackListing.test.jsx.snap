// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackListing component to render a game with 3 tracks 1`] = `
<DocumentFragment>
  <div>
    <table
      class="table table-bordered track-listing"
    >
      <thead>
        <tr>
          <th
            class="heading"
            colspan="6"
          >
            Track listing for Game 18-04-25-2: "Pot Luck"
          </th>
        </tr>
        <tr>
          <th
            class="number"
          >
            #
          </th>
          <th
            class="start-time"
          >
            Start
          </th>
          <th
            class="title"
          >
            Title
          </th>
          <th
            class="artist"
          >
            Artist
          </th>
          <th
            class="album"
          >
            Album
          </th>
          <th
            class="duration"
          >
            Duration
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="track green-theme"
          data-testid="track[2]"
        >
          <td
            class="number"
          >
            2
          </td>
          <td
            class="start-time"
          >
            00:40.0
          </td>
          <td
            class="title"
          >
            You Spin Me Round (Like a Record)
          </td>
          <td
            class="artist"
          >
            Dead Or Alive
          </td>
          <td
            class="album"
          >
            100 Hits 80s Essentials
          </td>
          <td
            class="duration"
          >
            00:30.0
          </td>
        </tr>
        <tr
          class="track green-theme"
          data-testid="track[3]"
        >
          <td
            class="number"
          >
            3
          </td>
          <td
            class="start-time"
          >
            01:11.0
          </td>
          <td
            class="title"
          >
            Mad World (From "Donnie Darko")
          </td>
          <td
            class="artist"
          >
            Tears For Fears
          </td>
          <td
            class="album"
          >
            Now That's What I Call Movies
          </td>
          <td
            class="duration"
          >
            00:30.0
          </td>
        </tr>
        <tr
          class="track green-theme"
          data-testid="track[4]"
        >
          <td
            class="number"
          >
            4
          </td>
          <td
            class="start-time"
          >
            01:42.1
          </td>
          <td
            class="title"
          >
            Why (From "Soup for One")
          </td>
          <td
            class="artist"
          >
            Carly Simon
          </td>
          <td
            class="album"
          >
            Now That's What I Call Movies
          </td>
          <td
            class="duration"
          >
            00:30.0
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
